hub.py

```diff
@@ -260,6 +260,22 @@ def setup_nfs_share(self):
             f'sudo mkdir -p {self.nfs_share_name} && sudo chown 1000:1000 {self.nfs_share_name} && sudo ls -ld {self.nfs_share_name}'
         ])
 
+    def check_hub_health(self, test_notebook, service_api_token):
+        """
+        After each hub gets deployed, validate that it 'works'.
+
+        Automatically create a temporary user, start their server and run a test notebook, making
+        sure it runs to completion. Stop and delete the test server at the end. Try this steps twice
+        before declaring it a failure. If any of these steps fails, immediately halt further
+        deployments and error out.
+        """
+        hub_url = f'https://{self.spec["domain"]}'
+        # Export the hub health check service as an env var so that jupyterhub_client can read it.
+        os.environ['JUPYTERHUB_API_TOKEN']=service_api_token
+        subprocess.check_call([
```

Comment:
We should probably also catch the exception here if it fails, and return True or False. This allows us to replace the internal implementation detail of this method later without having to modify all the callers.